shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1150,10 +1387,52 @@ - (BOOL)hasText {
 }
 
 - (void)insertText:(NSString*)text {
+  NSMutableArray* copiedRects = [[NSMutableArray alloc] initWithCapacity:[_selectionRects count]];
+  NSUInteger insertPosition = ((FlutterTextPosition*)_selectedTextRange.start).index - 1;
+  NSUInteger insertIndex = 0;
+  for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
+    NSUInteger rectPosition = [_selectionRects[i][4] unsignedIntegerValue];
+    if (rectPosition == insertPosition) {
+      insertIndex = i;
+      for (NSUInteger j = 0; j <= text.length; j++) {
+        [copiedRects addObject:@[
+          _selectionRects[i][0],
+          _selectionRects[i][1],
+          _selectionRects[i][2],
+          _selectionRects[i][3],
+          [NSNumber numberWithInt:rectPosition + j],
+        ]];
+      }
+    } else {
+      if (rectPosition > insertPosition) {
+        rectPosition = rectPosition + text.length;
+      }
+      [copiedRects addObject:@[
+        _selectionRects[i][0],
+        _selectionRects[i][1],
+        _selectionRects[i][2],
+        _selectionRects[i][3],
+        [NSNumber numberWithInt:rectPosition],
+      ]];
+    }
+  }
+
+  _selectionRects = copiedRects;
   _selectionAffinity = _kTextAffinityDownstream;
   [self replaceRange:_selectedTextRange withText:text];
 }
 
+- (UITextPlaceholder*)insertTextPlaceholderWithSize:(CGSize)size {
+  [_textInputDelegate insertTextPlaceholderWithSize:size withClient:_textInputClient];
+  _hasPlaceholder = YES;
+  return [[FlutterTextPlaceholder alloc] init];
```

Comment:
You shouldn't return values that aren't released by the autorelease pool (read as this PR won't be accepted unless that change is made).